tests/e2e/utils/data_fetcher/metrics.py

```diff
@@ -0,0 +1,71 @@
+from datetime import datetime, timedelta
+from typing import List, Optional
+
+import boto3
+from mypy_boto3_cloudwatch import CloudWatchClient
+from mypy_boto3_cloudwatch.type_defs import DimensionTypeDef, MetricDataResultTypeDef
+from retry import retry
+
+from tests.e2e.utils.data_builder import build_metric_query_data
+
+
+@retry(ValueError, delay=2, jitter=1.5, tries=10)
+def get_metrics(
+    namespace: str,
+    start_date: datetime,
+    metric_name: str,
+    dimensions: Optional[List[DimensionTypeDef]] = None,
+    cw_client: Optional[CloudWatchClient] = None,
+    end_date: Optional[datetime] = None,
+    period: int = 60,
+    stat: str = "Sum",
+) -> MetricDataResultTypeDef:
+    """Fetch CloudWatch Metrics
+
+    It takes into account eventual consistency with up to 10 retries and 1s jitter.
+
+    Parameters
+    ----------
+    namespace : str
+        Metric Namespace
+    start_date : datetime
+        Start window to fetch metrics
+    metric_name : str
+        Metric name
+    dimensions : Optional[List[DimensionTypeDef]], optional
+        List of Metric Dimension, by default None
+    cw_client : Optional[CloudWatchClient], optional
+        Boto3 CloudWatch low-level client (boto3.client("cloudwatch"), by default None
+    end_date : Optional[datetime], optional
+        End window to fetch metrics, by default start_date + 2 minutes window
+    period : int, optional
+        Time period to fetch metrics for, by default 60
+    stat : str, optional
+        Aggregation function to use when fetching metrics, by default "Sum"
+
+    Returns
+    -------
+    MetricDataResultTypeDef
+        _description_
```

Comment:
Sorry in advance if I missed something, but is something missing here?

Reply:
Great catch!! Leftover from the last PR